src-cpp/app/modules/shared/section_model.h

```diff
@@ -0,0 +1,64 @@
+#ifndef SECTION_MODEL_H
+#define SECTION_MODEL_H
+
+#include <QAbstractListModel>
+#include <QHash>
+#include <QVector>
+
+#include "section_item.h"
+
+namespace Shared
+{
+namespace Models
+{
+class SectionModel : public QAbstractListModel
+{
+    Q_OBJECT
+
+public:
+    enum ModelRole
+    {
+        Id = Qt::UserRole + 1,
+        SectionType,
+        Name,
+        AmISectionAdmin,
+        Description,
+        Image,
+        Icon,
+        Color,
+        HasNotification,
+        NotificationsCount,
+        Active,
+        Enabled,
+        Joined,
+        IsMember,
+        CanJoin,
+        CanManageUsers,
+        CanRequestAccess,
+        Access,
+        EnsOnly,
+        MembersModel,
+        PendingRequestsToJoinModel
+    };
+
+    explicit SectionModel(QObject* parent = nullptr);
+    ~SectionModel() = default;
+
+    QHash<int, QByteArray> roleNames() const;
+    virtual int rowCount(const QModelIndex&) const;
+    virtual QVariant data(const QModelIndex& index, int role) const;
```

Comment:
Why this methods is `virtual` if you `override` it?

Reply:
missed removing them, will remove :)